### PR TITLE
feat(streams): add authenticated PolyBolt price streams

### PR DESCRIPTION
**Pull request by kartojal**
Adds authenticated crypto, 60-second TWAP, and equity price streams through `AsyncSecureClient.subscribe()`, matching ts-sdk behavior with Python-native specs, Decimal prices, and UTC datetimes.

Connections are shared across subscriptions, with automatic reconnection and async context-manager cleanup. Integration workflows cover normal usage; focused unit tests cover failure and edge cases.

---

> [!NOTE]
> **Medium Risk**
> Introduces authenticated WebSocket price subscriptions and deprecates legacy public price specs; behavior changes for migrations (symbol format, USD vs USDT, topics) though old paths remain available.
> 
> **Overview**
> Adds **authenticated realtime price streaming** on `AsyncSecureClient.subscribe()` via `CryptoPriceSpec`, `CryptoTwapPriceSpec`, and `EquityPriceSpec`, backed by a new PolyBolt WebSocket stack (`realtime_ws_url`) that pools shared symbol filters (up to 64 per connection), waits for server acceptance, delivers **history** (`type="subscribe"`) then **live** (`type="update"`) events with `Decimal` prices and UTC `datetime`s, and handles auth, reconnect, and heartbeat.
> 
> **`AsyncPublicClient`** now rejects `PriceSpec` batches with `UserInputError`; legacy public RTDS specs (`CryptoPricesSpec`, `CryptoPricesChainlinkTwapSpec`, `EquityPricesSpec`) are **deprecated** but unchanged for existing callers. Subscription **handles** close more predictably (discard buffered events, shielded close, merged streams surface child errors sooner).
> 
> Docs (`streams.rst`), `examples/stream_realtime_prices`, integration/unit tests, and `typing-extensions` dependency support the new API.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 59963add7914a937fdce73d922881935c6a25dbf. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/polymarket/clients/async_secure.py [hunk 1]
```diff
@@ -1029,6 +1041,20 @@ async def revoke_session_key(
...
+    @overload
+    async def subscribe(
+        self, specs: CryptoTwapPriceSpec | Sequence[CryptoTwapPriceSpec], /
+    ) -> SubscriptionHandle[CryptoTwapPriceEvent]: ...
+    @overload
+    async def subscribe(
+        self, specs: EquityPriceSpec | Sequence[EquityPriceSpec], /
+    ) -> SubscriptionHandle[EquityPriceEvent]: ...
+    @overload
+    async def subscribe(self, specs: Sequence[PriceSpec], /) -> SubscriptionHandle[PriceEvent]: ...
```
Comment (cesarenaldi) [inline src/polymarket/clients/async_secure.py:1068]: Could we preserve only the requested event types for mixed subscriptions? Crypto plus equity currently returns `PriceEvent`, which also includes TWAP. Please narrow that result to `CryptoPriceEvent | EquityPriceEvent` and update the direct-client typing test to assert the precise union.

File: src/polymarket/streams/_specs.py [hunk 2]
```diff
@@ -177,6 +181,9 @@ class CryptoPricesChainlinkTwapSpec:
     ``window_seconds`` selects the 30-second or 60-second averaging window.
     Symbols are lowercase slash-delimited pairs such as ``btc/usd``. When
     ``symbols`` is omitted, the subscription receives every symbol.
+
+    Deprecated: use ``CryptoTwapPriceSpec`` for a fixed 60-second USD TWAP
```
Comment (cesarenaldi) [inline src/polymarket/streams/_specs.py:211]: Could we make these deprecations actionable through both editor diagnostics and migration examples?

Use `typing_extensions.deprecated` on the three legacy spec classes, keeping the warning concise. With `category=None`, supporting editors/type checkers can flag usage without adding runtime warnings. Put the before/after example and relevant caveats in each class’s docstring. For example:

```python
@deprecated(
    "Use CryptoTwapPriceSpec with AsyncSecureClient.",
    category=None,
)
@dataclass(frozen=True, slots=True, kw_only=True)
class CryptoPricesChainlinkTwapSpec:
    """Subscribe to legacy TWAP updates.

    Examples:
        Before:
            CryptoPricesChainlinkTwapSpec(
                window_seconds=60, symbols=["btc/usd"]
            )

        After, using AsyncSecureClient:
            CryptoTwapPriceSpec(symbols=["btcusd"])

    The replacement supports only 60-second TWAP.
    There is no replacement for the 30-second stream.
    """
    # Existing fields and implementation.
```

Please apply the same pattern to `CryptoPricesSpec` and `EquityPricesSpec`, documenting the relevant symbol, feed, and event-shape changes so the examples do not imply identical behavior. Since we support Python 3.11, declare `typing_extensions` as a direct dependency if we use this decorator.

Comment (kartojal) [reply src/polymarket/streams/_specs.py:211]: added the requested changes in the docstrings,

File: src/polymarket/clients/async_secure.py (hunk 1 above)
Comment (kartojal) [reply src/polymarket/clients/async_secure.py:1068]: updated!

File: src/polymarket/models/price_events.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    RATE_LIMITED = "rate_limited"
+    AUTH_REQUIRED = "auth_required"
+    AUTH_UNAVAILABLE = "auth_unavailable"
+    AUTH_EXPIRED = "auth_expired"
+    AUTH_ATTEMPTS = "auth_attempts"
+    AUTH_INVALID = "auth_invalid"
+
+
+class RealtimePricePoint(BaseModel):
+    """One exact USD price at a UTC instant."""
```
Comment (cesarenaldi) [inline src/polymarket/models/price_events.py]: Could we avoid describing all equity-feed and shared price values as USD? This feed also supports forex pairs such as `usdjpy` and `usdcad`, and the SDK preserves their original quote units. Please describe these values in the instrument’s quote currency, retaining USD-specific wording for the crypto and crypto TWAP streams.

Comment (kartojal) [reply src/polymarket/models/price_events.py]: good catch, on it

Comment (kartojal) [reply src/polymarket/models/price_events.py]: fixed

**Review by cesarenaldi — APPROVED** [review]
Verified the mixed-subscription typing, legacy deprecation guidance, and quote-currency documentation fixes. No outstanding findings from this review.

Lint, formatting, type checking, and the documentation build pass at this commit. CI is green. The implementation passed 2,458 non-live tests and both read-only live price-stream integration workflows during the preceding review; the final commit changes documentation and docstrings only.
